necromancyonline/necromancy-server
# Feature/item library

This PR adds a new Item Library source table and item library CSV to C:\Github\necromancy-server\Necromancy.Server\Client\Data\Settings. item_necromancy.csv is now obsolete. 

ItemType and Equiptype have been manually added as columns in the item library. the item library is based off hiraeths google docs file which pulled settings from Miku's GitHub.    Items from all clients, are in the library, but only the 'us steam client' items will load via NecDatabaseBuilder, as the match to itemLibrary.csv is performed on itemInfo.CSV  

ItemType, EquipType, and LoadEquipType Enums have been updated with accurate values.  they can be found in the models\itemModel folder.   they are used during initial DB build, and during equip/unequip operations

Equipment is now stored in the database, and will load on your character at character select screen and login to world.  Equipment in 'cloakRoom' will load across all characters per soul.

Blacklist has been un-commented and repaired.  structure is better documented to include timestamp, lock, and bounty.
Union invite/accept has been repaired to use the correct SQL select union query

Inventory has been moved from client to character.    Needed to leverage inventory class on character select screen before character was assigned to client.

Added some SQL queries (thanks for the assist noth!)
made a new CSV reader for itemLibrary

Tests:

- Delete BIN and re-load server.  expect this to take ~10 minutes.  building the ItemLibrary table takes a while. 
- Log in as ff/ff - confirm you have items equipped at character select.  should be a giant two handed sword.
- Check /stor  . you should have some items in there.   Move some stuff around between cloak room and inventory.  un-equip some things.  Log out back in.  changes should be reflected.
- create a new character.   log out and back in.  check /stor. you should see some items in there.
- check your blacklist.  should be 10 entries.
- Log in as 1234/1234 - Kadred and Xeno have new equipment.   Zakura is still using old loadequipment logic (blue dress avatar).
- as xeno, invite your new character to trade_union.  should work.

Known Issues:

- Avatar items don't render on char select or in game unless you un-equip and re-equip.  Will work on that before next PR.  
- Not all Items load the correct Icons.   e.g. /itm 210401 works, but 210405 doesn't.  210405 should use 210401s icon and model with alt texture overlay.  will work on that before next PR
- if you don't have any equipment equipped, you load with some default armor rendered.  Not quite ready to delete the 'old' loadequip functions. soon...
- if you die, your soul form shows full gear.   death/revive needs a lot of re-work. will make a separate feature branch for that.
